builder/proxmox/common/config.go

```diff
@@ -171,6 +172,58 @@ type vgaConfig struct {
 	Memory int    `mapstructure:"memory"`
 }
 
+// Allows passing through a host PCI device into the VM. For example, a graphics card
+// or a network adapter. Devices that are mapped into a guest VM are no longer available
+// on the host. A minimal configuration only requires either the `host` or the `mapping`
+// key to be specifed.
+//
+// Note: VMs with passed-through devices cannot be migrated.
+//
+// Example configuration (HCL):
+//
+//	```hcl
+//	pci_devices {
+//	host          = "0000:0d:00.1"
+//	pcie          = false
+//	device_id     = "1003"
+//	legacy_igd    = false
+//	mapping       = "someNic"
+//	mdev          = "some-model"
+//	hide_rombar   = false
+//	romfile       = "vbios.bin"
+//	sub_device_id = ""
+//	sub_vendor_id = ""
+//	vendor_id     = "15B3"
+//	x_vga         = false
+//	}
+//	```
+type pciDeviceConfig struct {
+	// The PCI ID of a host’s PCI device or a PCI virtual function. You can us the `lspci` command to list existing PCI devices. Either this or the `mapping` key must be set.
+	Host string `mapstructure:"host" required:"true"`
+	// Override PCI device ID visible to guest.
+	DeviceID string `mapstructure:"device_id"`
+	// Pass this device in legacy IGD mode, making it the primary and exclusive graphics device in the VM. Requires `pc-i440fx` machine type and VGA set to `none`. Defaults to `false`.
+	LegacyIGD bool `mapstructure:"legacy_igd"`
+	// The ID of a cluster wide mapping. Either this or the `host` key must be set.
+	Mapping string `mapstructure:"mapping" required:"true"`
```

Comment:
I wonder given how the comments are phrased, are configurations with either `host` or `mapping` accepted? My understanding with `required` is that both options will be required if we note them as `required:"true"`.
In cases like these, I would suggest leaving `required` as false, and checking that one or the other is set, and not both.

Reply:
I believe that your tests won't catch this since it's related to parsing configurations, this may be tested via acceptance tests, but AFAIK we don't have those setup for proxmox (we'd need some infrastructure for this).
In the meantime, could you manually check the point I bring here? I would like to make sure this won't be an issue before merging this PR.

Reply:
Looks like you're right about this - [both cannot be set](https://github.com/proxmox/qemu-server/blob/742da46a44f7d882620ae0d8d8cc08ab7b3f8dcc/PVE/QemuServer/PCI.pm#L428) 👍